src/py/flwr/server/superlink/fleet/vce/vce_api.py

```diff
@@ -42,21 +46,143 @@ def _register_nodes(
     return nodes_mapping
 
 
-# pylint: disable=too-many-arguments,unused-argument
+# pylint: disable=too-many-arguments,too-many-locals
+async def worker(
+    app: Callable[[], ClientApp],
+    queue: "asyncio.Queue[TaskIns]",
+    node_states: Dict[int, NodeState],
+    state_factory: StateFactory,
+    nodes_mapping: NodeToPartitionMapping,
+    backend: Backend,
+) -> None:
+    """Get TaskIns from queue and pass it to an actor in the pool to execute it."""
+    state = state_factory.state()
+    while True:
+        try:
+            task_ins: TaskIns = await queue.get()
+            node_id = task_ins.task.consumer.node_id
+
+            # Register and retrive runstate
+            node_states[node_id].register_context(run_id=task_ins.run_id)
+            context = node_states[node_id].retrieve_context(run_id=task_ins.run_id)
+
+            # Convert TaskIns to Message
+            message = message_from_taskins(task_ins)
+            # Replace node-id with data partition id
+            message.metadata.dst_node_id = nodes_mapping[node_id]
+
+            # Let backend process message
+            out_mssg, updated_context = await backend.process_message(
+                app, message, context
+            )
+
+            # Update Context
+            node_states[node_id].update_context(
+                task_ins.run_id, context=updated_context
+            )
+
+            # Undo change node_id for partition choice
+            out_mssg.metadata._src_node_id = (  # pylint: disable=protected-access
+                task_ins.task.consumer.node_id
+            )
+            # Convert to TaskRes
+            task_res = message_to_taskres(out_mssg)
+            # Store TaskRes in state
+            state.store_task_res(task_res)
+
+        except asyncio.CancelledError as e:
+            log(DEBUG, "Async worker: %s", e)
+            break
+
+        except Exception as ex:  # pylint: disable=broad-exception-caught
+            # pylint: disable=fixme
+            # TODO: gen TaskRes with relevant error, add it to state_factory
```

Comment:
Let's remove the TODO here and make a separate note of it - it's related to #2846